stl/src/cthread.cpp

```diff
@@ -99,9 +99,16 @@ _Thrd_id_t _Thrd_id() { // return unique id for current thread
 }
 
 unsigned int _Thrd_hardware_concurrency() { // return number of processors
-    SYSTEM_INFO info;
-    GetNativeSystemInfo(&info);
-    return info.dwNumberOfProcessors;
+    DWORD_PTR process_affinity;
+    DWORD_PTR system_affinity;
+    if (GetProcessAffinityMask(GetCurrentProcess(), &process_affinity, &system_affinity))
+#ifdef _WIN64
+        return __builtin_popcountll(process_affinity);

```

Comment:
@barcharcraz Did this call need to be guarded for the CPU having that instruction?

Reply:
I've seen it in std::popcount, so if it should be guarded, std::popcount should be fixed too...

Reply:
Yeah, there was some back and forth with the backend team on that and I'm not sure that landed. Since that was Charlie's feature I'd just like him to confirm.

Reply:
Since this is separately compiled, can we just use `<bit>`?

Reply:
`<bit>` is C++20, apparently this code is not compiled to have C++20 features included.

Would use of `std::bitset` be better then?
```
    if (GetProcessAffinityMask(GetCurrentProcess(), &process_affinity_mask, &system_affinity_mask)) {
        std::bitset<sizeof(DWORD_PTR) * CHAR_BIT> process_affinity_mask_bits(
            static_cast<unsigned long long>(process_affinity_mask));
        return static_cast<unsigned int>(process_affinity_mask_bits.count());
    } else {
        return 0;
    }
```

Reply:
Figured out why `<bit>` does not work. Need to define `__cpp_lib_bitops` to make it work.

I see that this define is disabled in `yvals_core.h`.

```
#if defined(__clang__) || defined(__EDG__)
#define __cpp_lib_bitops 201907L
#else // ^^^ Clang and EDG / MSVC vvv
// a future MSVC update will embed CPU feature detection into <bit> intrinsics
// TRANSITION, VSO-1020212
#endif // defined(__clang__) || defined(__EDG__)

```

So, although `__builtin_popcount` / `__builtin_popcountll` seem to work, they are not yet published. My observation also shows that there's no CPU check even for `/arch:IA32`

Waiting for response about `__builtin_popcount` / `__builtin_popcountll` usage then.

Reply:
Actually let's go with `<bitset>` for now, no much point in saving on bit operation, if there's a kernel call anyway.